### PR TITLE
Use Spring 4.3 composed annotations for @RequestMapping

### PR DESCRIPTION
**Pull request by nkolosnjaji**
Update code to use new Spring 4.3 composed annotation in rest packages in server and entity generator. This will remove some code and make it clearer for developers to find methods.

see http://docs.spring.io/spring/docs/current/spring-framework-reference/html/mvc.html#mvc-ann-requestmapping-composed

### CODE REVIEW

**Comment by jdubois** [comment]
This is a huge change -> please the process is to open a ticket first so we can discuss this.
But this is a good change, +1 from me already :-)

**Comment by jdubois** [comment]
And can we make it even smaller by removing the "value="?
Sorry I'm taking a flight, no time to check the documentation

**Comment by nkolosnjaji** [comment]
yes @jdubois, attribute "value" can be removed in some places where it is single attribute. I can try to reduce it even more by removing "produces = MediaType.****" stuff and test it, if you think is ok.

P.S. I created a issue (feature)

**Comment by jdubois** [comment]
Awesome!!!

**Comment by sendilkumarn** [comment]
Good one 👍

**Comment by deepu105** [comment]
👍 frommeas well, its cleaner and more readable

**Comment by deepu105** [comment]
@nkolosnjaji yes go ahead

**Comment by jdubois** [comment]
Yes that's really looking cool!!

**Comment by stevehouel** [comment]
Looks really nice and easy to understand. Thx @nkolosnjaji

File: generators/server/templates/src/main/java/package/web/rest/_AccountResource.java [hunk 1]
```diff
@@ -140,9 +133,7 @@ public String isAuthenticated(HttpServletRequest request) {
      * @param userDTO the current user information
      * @return the ResponseEntity with status 200 (OK), or status 400 (Bad Request) or 500 (Internal Server Error) if the user couldn't be updated
      */
-    @RequestMapping(value = "/account",
-        method = RequestMethod.POST,
-        produces = MediaType.APPLICATION_JSON_VALUE)
+    @PostMapping(path = "/account")
```
Comment (mehmet-aslan) [inline generators/server/templates/src/main/java/package/web/rest/_AccountResource.java]: path can be removed here

Comment (nkolosnjaji) [inline generators/server/templates/src/main/java/package/web/rest/_AccountResource.java]: yes, I will modify this... 

p.s. What about ng2 branch ?

**Comment by deepu105** [comment]
Not required for ng2. As it will be merged to master

Thanks & regards,
Deepu

**Comment by jdubois** [comment]
I'm really looking forward to use this!!

**Comment by PierreBesson** [comment]
Really nice :+1:

**Pull request merged**